evolver/app/routers/hardware.py

```diff
@@ -80,34 +87,50 @@ def get_calibrator_actions(hardware_name: str, request: Request):
 
 
 # Dispatch an action to the calibration procedure
-@router.post("/{hardware_name}/calibrator/dispatch")
+@router.post("/{hardware_name}/calibrator/procedure/dispatch")
 def dispatch_calibrator_action(request: Request, hardware_name: str = Path(...), action: dict = Body(...)):
     hardware_instance = get_hardware_instance(request, hardware_name)
     calibrator = hardware_instance.calibrator
     if not calibrator:
         raise HTTPException(status_code=404, detail=f"Calibrator not found for '{hardware_name}'")
 
     calibration_procedure = calibrator.calibration_procedure
+
     action_to_dispatch = next((a for a in calibration_procedure.get_actions() if a.name == action["action_name"]), None)
 
     if not action_to_dispatch:
         raise HTTPException(status_code=404, detail=f"Action '{action['action_name']}' not found")
 
     try:
-        payload = action["payload"]
+        payload = action.get("payload", {})
         new_state = calibration_procedure.dispatch(action_to_dispatch, payload)
     except pydantic.ValidationError as e:
         raise HTTPException(status_code=422, detail=f"Invalid payload: {e.errors()}")
 
     return {"state": new_state}
 
 
-# Get the current state of the calibrator for a hardware
-@router.get("/{hardware_name}/calibrator/state")
+# Get the current state of the calibration procedure
+@router.get("/{hardware_name}/calibrator/procedure/state")
 def get_calibrator_state(hardware_name: str, request: Request):
     hardware_instance = get_hardware_instance(request, hardware_name)
     calibrator = hardware_instance.calibrator
     if not calibrator:
-        raise CalibratorNotFoundError
+        raise HTTPException(status_code=404, detail=f"Calibrator not found for '{hardware_name}'")
+    calibration_procedure = calibrator.calibration_procedure
+    return calibration_procedure.get_state()
+
+
+@router.get("/{hardware_name}/calibrator/data")
```

Comment:
This is the persisted CalibrationData, this is populated only once a Procedure reaches a point when its measured_data can be meaningfully persisted.